beetsplug/playlist.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+# This file is part of beets.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+import os
+import beets
+
+
+class PlaylistQuery(beets.dbcore.FieldQuery):
+    """Matches files listed by a playlist file.
+    """
+    def __init__(self, field, pattern, fast=True):
+        super(PlaylistQuery, self).__init__(field, pattern, fast)
+        config = beets.config['playlist']
+
+        # Get the full path to the playlist
+        if os.path.isabs(beets.util.syspath(pattern)):
```

Comment:
That would introduce a race condition in case the file is deleted between the `isfile` and the `open` call. With proper error handling this wouldn't be too bad, but I restructured the code a bit so that `isfile` is not needed at all.

Reply:
Looks perfect. :+1: